server/application/application.go

```diff
@@ -211,7 +211,7 @@ func (s *Server) Create(ctx context.Context, q *application.ApplicationCreateReq
 		reflect.DeepEqual(existing.Finalizers, a.Finalizers)
 
 	if equalSpecs {
-		return existing, nil
+		return existing, status.Errorf(codes.AlreadyExists, "application '%s' already exists and no changes needed", q.Application.Name)
```

Comment:
I'm a bit hesitant to make a breaking API change. Maybe this is a change for 3.0?

Reply:
An alternative to make this non-breaking might be to compare the app returned by the server with the app sent in the request from the CLI. I had tried that before, but couldn't quite get it to work. I can look into it again.

Reply:
That would be nice... I'd consider it a low-priority issue, but if you figure it out, that'd be great!

Reply:
Added in some checking logic to the CLI so the server doesn't return an error.